### ISSUE TITLE
[SonicFrontiers] Most Mods Stopped Working

### ISSUE THREAD

**Issue by TrepeSerafin** [thesupersonic16/HedgeModManager#695]
Most of my mods suddenly began to not function. I downgraded HMM and made a new mods folder with 2 mods that I'm certain work without issue, and then I got this exception.

[HMM_Snapshot_00082023163707.txt](https://github.com/thesupersonic16/HedgeModManager/files/12412870/HMM_Snapshot_00082023163707.txt)

HMM Info:
```
    Version: 7.10-1
    Args: 
    StartDir: D:\SteamLibrary\steamapps\common\SonicFrontiers\
    Process Level: User
    Game: Sonic Frontiers
    GameInstall: Sonic Frontiers
    Update Status: Up To Date
```

Exception:
```
    Type: NullReferenceException
    Message: Object reference not set to an instance of an object.
    Source: HedgeModManager
    Function: System.String Resolve(System.String)
    StackTrace: 
       at HedgeModManager.ModsDB.Resolve(String name) in D:\GitHub\HedgeModManager\HedgeModManager\ModsDB.cs:line 541
       at HedgeModManager.CodeCompiler.PreProcessor.TextProcessor.Process(String text)
       at HedgeModManager.CodeCompiler.SyntaxTreeEx.ProcessText(String text, TextProcessor& preprocessor, IIncludeResolver includeResolver)
       at HedgeModManager.CodeCompiler.SyntaxTreeEx.Parse(String text, IIncludeResolver includeResolver)
       at HedgeModManager.CodeCompiler.CodeProvider.GetLoadAssemblies[TCollection](TCollection sources, IIncludeResolver includeResolver, String[] lookupPaths)
       at HedgeModManager.CodeCompiler.CodeProvider.CompileCodes[TCollection](TCollection sources, Stream resultStream, IIncludeResolver includeResolver, String[] loadPaths)
       at HedgeModManager.CodeCompiler.CodeProvider.CompileCodes[TCollection](TCollection sources, String assemblyPath, IIncludeResolver includeResolver, String[] loadsPaths)
       at HedgeModManager.ModsDB.<SaveDB>d__30.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at HedgeModManager.MainWindow.<SaveModsDB>d__42.MoveNext() in D:\GitHub\HedgeModManager\HedgeModManager\UI\MainWindow.xaml.cs:line 655
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at HedgeModManager.MainWindow.<SaveConfig>d__54.MoveNext() in D:\GitHub\HedgeModManager\HedgeModManager\UI\MainWindow.xaml.cs:line 986
    InnerException: 
```

**Comment by TrepeSerafin** [comment]
Update:
I'm no longer getting the exception after updating community code.
Many mods that didn't work with the latest version of HMM began to work again as well.
However, some of my mods are still not loading properly despite looking fine in debug console.

![image](https://github.com/thesupersonic16/HedgeModManager/assets/142684252/ba8c8e1e-f8b1-4abc-b52e-cf6548368ecd)

**Comment by thesupersonic16** [comment]
Can you explain what you mean by "not loading properly"? Like is all files shown have no effect and all you are seeing is the vanilla files?

**Comment by TrepeSerafin** [comment]
Yes. Some loaded mods just have no effect. It appears I'm not the only one having this problem, but it's affecting very few users.

**Comment by TrepeSerafin** [comment]
The issue is seemingly fixed with version 7.11.
